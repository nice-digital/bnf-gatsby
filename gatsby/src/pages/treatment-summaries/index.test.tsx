import { render, waitFor, screen } from "@testing-library/react";

import TreatmentSummariesIndexPage, {
	type TreatmentSummariesIndexPageProps,
} from "./";

const props: TreatmentSummariesIndexPageProps = {
	data: {
		allTreatmentSummaries: {
			letters: [
				{
					letter: "a",
					links: [
						{
							title: "Acne",
							slug: "acne",
						},
						{
							title: "Acute coronary syndromes",
							slug: "acute-coronary-syndromes",
						},
					],
				},
				{
					letter: "b",
					links: [
						{
							title: "Bacillus Calmette-GuÃ©rin vaccine",
							slug: "bacillus-calmette-guerin-vaccine",
						},
						{
							title: "Beta-adrenoceptor blocking drugs",
							slug: "beta-adrenoceptor-blocking-drugs",
						},
					],
				},
			],
		},
	},
};

describe("TreatmentSummariesIndexPage", () => {
	it("should set page title", async () => {
		render(<TreatmentSummariesIndexPage {...props} />);
		await waitFor(() => {
			expect(document.title).toStartWith("Treatment summaries A to Z");
		});
	});

	it("should set page meta description", async () => {
		render(<TreatmentSummariesIndexPage {...props} />);
		await waitFor(() => {
			expect(
				document
					// eslint-disable-next-line testing-library/no-node-access
					.querySelector("meta[name='description']")
					?.getAttribute("content")
			).toEqual(
				"Browse the complete list of treatment summaries, alphabetically."
			);
		});
	});

	it("should match snapshot for page body", () => {
		render(<TreatmentSummariesIndexPage {...props} />);
		expect(screen.getByRole("main")).toMatchSnapshot();
	});
});
