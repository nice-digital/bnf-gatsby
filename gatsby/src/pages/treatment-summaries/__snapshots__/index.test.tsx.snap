// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreatmentSummariesIndexPage should match snapshot for page body 1`] = `
<main
  class="main"
>
  <div
    class="container"
  >
    <nav
      aria-label="Breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="https://www.nice.org.uk/"
          >
            NICE
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            BNF
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            Treatment summaries A to Z
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "Treatment summaries A to Z"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading"
      >
        Treatment summaries
         A to Z
      </h1>
    </div>
    <ol
      style="display: flex; list-style: none; margin-left: 0px;"
    >
      <li>
        <a
          href="#a"
          style="padding: .5rem;"
        >
          A
        </a>
      </li>
      <li>
        <a
          href="#b"
          style="padding: .5rem;"
        >
          B
        </a>
      </li>
    </ol>
    <ol
      aria-label="Letters A to Z with matching links to Treatment summaries"
    >
      <li>
        <h2
          aria-label="Treatment summaries starting with letter 'A'"
          id="a"
          tabindex="-1"
        >
          A
        </h2>
        <ol
          aria-labelledby="a"
        >
          <li>
            <a
              href="/treatment-summaries/acne/"
            >
              Acne
            </a>
          </li>
          <li>
            <a
              href="/treatment-summaries/acute-coronary-syndromes/"
            >
              Acute coronary syndromes
            </a>
          </li>
        </ol>
      </li>
      <li>
        <h2
          aria-label="Treatment summaries starting with letter 'B'"
          id="b"
          tabindex="-1"
        >
          B
        </h2>
        <ol
          aria-labelledby="b"
        >
          <li>
            <a
              href="/treatment-summaries/bacillus-calmette-guerin-vaccine/"
            >
              Bacillus Calmette-Guérin vaccine
            </a>
          </li>
          <li>
            <a
              href="/treatment-summaries/beta-adrenoceptor-blocking-drugs/"
            >
              Beta-adrenoceptor blocking drugs
            </a>
          </li>
        </ol>
      </li>
    </ol>
  </div>
  <div
    class="wrapper"
  >
    <nav
      aria-labelledby="back-to-top-link"
      class="nav"
    >
      <a
        class="anchor"
        href="#top"
        id="back-to-top-link"
      >
        <div
          class="container"
        >
          <svg
            aria-hidden="true"
            class="icon"
            fill="none"
            focusable="false"
            height="1em"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M478.152 338.576c0 10.091-3.525 18.661-10.576 25.712l-21.424 21.424c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864L255.576 247.136 116.712 385.712c-6.859 7.243-15.429 10.864-25.712 10.864s-18.853-3.621-25.712-10.864l-21.424-21.424C36.621 357.429 33 348.859 33 338.576c0-10.091 3.621-18.757 10.864-26l186-186c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576l185.712 186c7.243 7.243 10.864 15.909 10.864 26z"
              fill="currentColor"
            />
          </svg>
           Back to top
        </div>
      </a>
    </nav>
  </div>
</main>
`;
